### ISSUE TITLE
Why no more BLIP?

### ISSUE THREAD

**Issue by MasterScrat** [Linaqruf/kohya-trainer#67]
Hello, great work on the notebooks! 🙌

I see you replaced `BLIP` with `Microsoft/GIT`. I am curious what was the reason for that? does it perform better? any license issues? We've been using `BLIP` in the past and it performed well.

**Comment by Fannovel16** [comment]
GIT is the successor of BLIP. I find it petforms better than BLIP, but large models (e.g. the default git-large-textcaps) is slower than BLIP though.

**Comment by MasterScrat** [comment]
Interesting, thanks!

Do you know how it compares to BLIP2 which was released a few days ago?
https://huggingface.co/spaces/Salesforce/BLIP2

**Comment by Fannovel16** [comment]
I have not tested BLIP2 but iirc it is a LLM so it is slower than GIT. However, I have found some images for comparision:
![](https://cdn.discordapp.com/attachments/1054180647196958770/1071752951686766622/image.png)
![](https://cdn.discordapp.com/attachments/1054180647196958770/1071752848817262672/image.png)

**Comment by Fannovel16** [comment]
I think it is better to use git-large-r-textcaps instead of git-large-textcaps because Microsoft retrained it with captions whose special tokens are removed.

**Comment by Fannovel16** [comment]
Also you can use this space to compare them:
https://huggingface.co/spaces/nielsr/comparing-captioning-models

**Comment by GeneralAwareness** [comment]
I use a bunch to caption my stuff and I am finding BlIP-2 to actually being used more. I take BlIP-2 and add a little of git (one of them) to it as it seems if they are combined manually you will get a superior caption (I only do real world styles).

**Comment by jvkap** [comment]
Blip 2 is good, but it is a gigantic resource eater. It is not worth . The ones that are considered light and good today are Git and Coca which are of equivalent quality.

**Comment by GeneralAwareness** [comment]
> Blip 2 is good, but it is a gigantic resource eater. It is not worth . The ones that are considered light and good today are Git and Coca which are of equivalent quality.

I disagree as I find I use Blip 2 far more than those other two as I have two sites up and put the same image in both.  Resource hog, or not, I rather use one that is more accurate when I train than to waste time having to find out it sucked, and I am forced to redo my training.

**Comment by Linaqruf** [comment]
Git-large-textcaps is much better than Blip (v1), providing more accurate and detailed captions. It also handles illustration captioning, such as anime and cartoons, well.

However, after a week with Git, I found that it is too slow for T4 users, especially if the dataset has high resolution. As a result, we will switch back to using Blip for captioning in the next update. I have also discovered that using Blip with beam search enabled yields similar caption quality.

Although Blip 2 is good, it requires too many resources just for inference and is too large.
